### DISCUSSION TITLE
How does the train_able constellation work?

### DISCUSSION THREAD

**Discussion by xiaobaijh** [NVlabs/sionna#708] [General]
Hello,

I’ve been exploring Sionna and am particularly interested in the feature that allows the constellation diagram to be trainable. I noticed that this functionality is implemented by defining the constellation points as tf.Variable, which makes them trainable.

However, I am a bit confused about the mechanism behind this. Specifically, it seems that the constellation points aren’t directly connected to the neural network during training, if we dont use  neural network in the receiver or transmitter . Could you clarify how the optimization process still updates the constellation points, given this apparent lack of connection to the network? Are there implicit dependencies or specific techniques used that enable the constellation diagram to be updated during training?

Looking forward to your insights!

Thank you for your time and this great tool.

Best regards,

**Comment by SebastianCa** [comment]
Hi @xiaobaijh,  

This is explained in [tutorial notebook 2](https://nvlabs.github.io/sionna/examples/Sionna_tutorial_part2.html). The main idea is to use gradient descent to optimize the constellation points to minimize the loss (e.g., the BCE loss).  

Both the `Mapper` and `Demapper` have access to the constellation object (i.e., the `tf.Variable`), and TensorFlow automatically identifies the trainable variables during backpropagation through the computational graph.  

I'll move this thread to discussions as it is not an issue with Sionna.
